NosTests/IntegrationTests/EventProcessorIntegrationTests+InlineMetadata.swift

```diff
@@ -0,0 +1,45 @@
+import XCTest
+
+extension EventProcessorIntegrationTests {
+    @MainActor func test_parse_gets_inline_metadata() throws {
+        // Arrange
+        let jsonData = try jsonData(filename: "text_note_with_media_metadata")
+        let jsonEvent = try JSONDecoder().decode(JSONEvent.self, from: jsonData)
+
+        // Act
+        let parsedEvent = try XCTUnwrap(
+            try EventProcessor.parse(jsonEvent: jsonEvent, from: nil, in: testContext, skipVerification: true)
+        )
+
+        let url = "https://image.nostr.build/6ffcd8dadcdf7a54a7f3d7bafb958d4d21040820b479db1047135d0cbfd24a95.png"
+        let inlineMetadataTag = try XCTUnwrap(parsedEvent.inlineMetadata?[url])
+        XCTAssertEqual(inlineMetadataTag.url, url)
+        XCTAssertEqual(inlineMetadataTag.dimensions, CGSize(width: 746, height: 443))
+    }
+
+    @MainActor func test_parse_multiple_inline_metadata() throws {
+        // Arrange
+        let jsonData = try jsonData(filename: "text_note_multiple_media")
+        let jsonEvent = try JSONDecoder().decode(JSONEvent.self, from: jsonData)
+
+        // Act
+        let parsedEvent = try XCTUnwrap(
+            try EventProcessor.parse(jsonEvent: jsonEvent, from: nil, in: testContext, skipVerification: true)
+        )
+
+        // Assert
+        let inlineMetadata = try XCTUnwrap(parsedEvent.inlineMetadata)
+
+        let squareImageURL =
+            "https://image.nostr.build/70f1f360919cbc044cfca6cc0d0ba1a420632c4828f7d22082d3463f33f06d7b.jpg"
+        let squareImageMetadata = try XCTUnwrap(inlineMetadata[squareImageURL])
+        XCTAssertEqual(squareImageMetadata.dimensions, CGSize(width: 854, height: 854))
+        XCTAssertEqual(squareImageMetadata.orientation, .landscape)
+
+        let portraitImageURL = 
+            "https://image.nostr.build/2787ad495941dfb068fbff77f087513095a3f981f9697fcb9e51c052c6198090.jpg"
+        let portraitImageMetadata = try XCTUnwrap(inlineMetadata[portraitImageURL])
+        XCTAssertEqual(portraitImageMetadata.dimensions, CGSize(width: 1263, height: 3985))
+        XCTAssertEqual(portraitImageMetadata.orientation, .portrait)
+    }
```

Comment:
It would be nice to have a test that verifies that things don't break on invalid data, like two imeta tags for the same URL. I assume in that sort of case we would just pick one arbitrarily.

Reply:
Yes! I'll add a test for that!

Reply:
Oh wow -- did you know it would crash if there are two `imeta` tags with the same URL? If not, I like your subtle suggestion to write a test. If so, bravo for seeing that just by looking at the code. Either way, I'm impressed. 😄

Reply:
Added the test, which showed the crash, which I subsequently fixed. The code is less pretty but more stable, so I'd call it a win.

Reply:
😆 I did not know it would crash